### DISCUSSION TITLE
shouldFocusError doesn't work with Controller?

### DISCUSSION THREAD

**Discussion by dwiyatci** [react-hook-form/react-hook-form#3112] [General] [answered]
I've read this note in the docs:
> Note: only registered fields with `ref` will work. Custom register inputs do not apply. eg: `register({ name: 'test' }) // doesn't work`

I just wanted to confirm my understanding on that sentence ('cause I can't believe what I read hahah), so a yes/no answer would suffice. 😃 I also made a CSB for this: https://codesandbox.io/s/react-hook-form-v6-errormessage-forked-7n6es?file=/src/index.js

So, `shouldFocusError` _really_ doesn't work with `Controller`, right?

**Comment by bluebill1049** [comment, marked as answer]
u mean `shouldFocusError`?

**Reply by dwiyatci** [reply]
right.  🤦🏻

**Reply by bluebill1049** [reply]
you need to look at onFocus Props at Contoller

**Reply by dwiyatci** [reply]
Aha! Interesting. I guess that'll do (I simply missed that section). Thaanks Bill!!

**Reply by zhushenhang** [reply]
@dwiyatci Same question. 
Are you using material-ui?

**Reply by dwiyatci** [reply]
@zhushenhang Sorry, I kinda missed to reply your message. 😳 

No, I'm not. I'm using proprietary UI widgets. But I believe you could take the advantage of this newly released `ref` from `Controller` for focus management (it's not that perfect yet though, e.g. focus might not be ordered in case of multiple errors): https://github.com/react-hook-form/react-hook-form/blob/master/CHANGELOG.md#6100---2020-10-31

**Reply by bluebill1049** [reply]
thanks @dwiyatci

**Reply by zhushenhang** [reply]
@dwiyatci thanks bro .
